feat(unity): Automated symbol upload (#4462)
getsentry/sentry-docs: src/platforms/unity/native-support.mdx

## Debug Symbols

> Sentry requires [debug information files](/platforms/android/data-management/debug-files/) to symbolicate your crash logs. The Unity SDK provides an automated upload functionality for those symbol files that relies on the [sentry-cli](/product/cli/). We included the executables for Windows, macOS, and Linux inside the Unity SDK package.

Reviewer: ```suggestion
Sentry requires [debug information files](/platforms/android/data-management/debug-files/) to symbolicate your crash logs. The Unity SDK provides an automated upload functionality for those symbol files that rely on the [sentry-cli](/product/cli/). We've included the executables for Windows, macOS, and Linux inside the Unity SDK package.
```

The automated debug symbols upload is enabled by default but requires configuration. Go to **Tools > Sentry > Editor** to enter the [Auth Token](https://sentry.io/api/), Org Slug, and the Project Name. Note that the Unity SDK creates a file at `Assets/Plugins/Sentry/SentryCliOptions.asset` to store the configuration, that should not be made publicly available.

### iOS - dSYM and Bitcode

Debug information files on the iOS platform are called dSYM, and the way to obtain them differs depending on whether `Enable Bitcode` is set to `true` in your Xcode project.

> For Sentry to symbolicate your crash logs and with `bitcode` enabled we need two types of files:

Reviewer: ```suggestion
For Sentry to symbolicate your crash logs and with `bitcode` enabled, we need two types of files:
```

1. `dSYM` files available only **after** App Store Connect finishes processing the build

> 2. `BCSymbolMap` files that get created during the archiving process

Reviewer: ```suggestion
2. `BCSymbolMap` files that are created during the archiving process
```

> The automated symbol upload will take care of the `BCSymbolMap` files by processing them during the archiving process. To provide the dSYM files to Sentry you can either set up the [App Store Connect Integration](/platforms/apple/guides/ios/dsym/#bitcode-appstore) so Sentry can fetch them for you or download and then upload them [manually using sentry-cli](/platforms/apple/guides/ios/dsym/#bitcode-sentrycli).

Reviewer: ```suggestion
The automated symbol upload will take care of the `BCSymbolMap` files by processing them during the archiving process. To provide the dSYM files to Sentry, you can either set up the [App Store Connect Integration](/platforms/apple/guides/ios/dsym/#bitcode-appstore) so Sentry can fetch them for you or download and then upload them [manually using sentry-cli](/platforms/apple/guides/ios/dsym/#bitcode-sentrycli).
```

> With `bitcode` disabled the automated symbols upload will pick up the `dSYM` files at the end of the build process without further action required.

Reviewer: ```suggestion
With `bitcode` disabled, the automated symbols upload will pick up the `dSYM` files at the end of the build process without further action required.
```
